### ISSUE TITLE
Final Loss equation

### ISSUE THREAD

**Issue by hardianlawi** [yaringal/multi-task-learning-example#1]
Hi @yaringal !

I have read the paper and it is really amazing. Thanks for your team's hard-work!

However, I have a question regarding the final equation and also your keras implementation.

![image](https://user-images.githubusercontent.com/22806872/34601530-5fcb66f6-f237-11e7-82a4-d249d20646fc.png)

The equation above has 1/2 multiplied by the loss, but you didn't include it in the keras implementation.

I tried experimenting on it, and included 1/2 in the loss function, but it couldn't converge. I am wondering if the problem is in the paper or the keras implementation, because if I exclude the 1/2, it converges to the ground-truth std.

Best regards,

Hardian

**Comment by yaringal** [comment]
Please have a look at the updated NIPS version of the paper which has some typos corrected.
Yarin

**Comment by hardianlawi** [comment]
hi @yaringal thanks for your reply

Are you referring to [this](papers.nips.cc/paper/by-source-2017-2871)? I can't seem to find the one with the same title

**Comment by yaringal** [comment]
Ah apologies this is not updated online yet. It will be soon.
The equation above has a type - the implementation is correct.
Yarin

**Comment by divyansha** [comment]
According to [the most recent version]( https://arxiv.org/pdf/1705.07115.pdf), the losses should still be multiplied by 1/2:
<img width="311" alt="screen shot 2019-01-14 at 6 00 26 pm" src="https://user-images.githubusercontent.com/6996908/51112979-8fafd580-1826-11e9-8807-a0b1f578084e.png">

Why is a 1/2 missing from your code?

**Comment by divyansha** [comment]
I did the math from scratch and it does seem like there is agreement between code and the most recent version of paper.

Apologies for the trouble.

**Comment by John1231983** [comment]
Here you are https://arxiv.org/pdf/1703.04977.pdf

![Screenshot from 2019-03-13 10-09-14](https://user-images.githubusercontent.com/24875971/54285329-21626780-4578-11e9-84cc-d7cbcc9a8c89.png)

I am implementing it in pytorch. If you have done it, please share it

**Comment by John1231983** [comment]
@hardianlawi : I reproduced the result from keras using pytorch. YOu can look at https://colab.research.google.com/drive/1_zsmQguerz0iy0J9Uu2Cs7oEHhj0QoXH

However, the sigma2 does not work. Do you have any suggestion @yaringal ?

**Comment by HaobingLiu** [comment]
> Hi @yaringal !
> 
> I have read the paper and it is really amazing. Thanks for your team's hard-work!
> 
> However, I have a question regarding the final equation and also your keras implementation.
> 
> ![image](https://user-images.githubusercontent.com/22806872/34601530-5fcb66f6-f237-11e7-82a4-d249d20646fc.png)
> 
> The equation above has 1/2 multiplied by the loss, but you didn't include it in the keras implementation.
> 
> I tried experimenting on it, and included 1/2 in the loss function, but it couldn't converge. I am wondering if the problem is in the paper or the keras implementation, because if I exclude the 1/2, it converges to the ground-truth std.
> 
> Best regards,
> 
> Hardian

I also have the same question. I think a 1/2 is missing according to the code. Do you know why?

**Comment by Banyueqin** [comment]
@John1231983 
you need to fix the code
![webwxgetmsgimg](https://user-images.githubusercontent.com/32731721/60170161-82b5fa00-983a-11e9-96fa-52956aef3b5e.jpeg)

**Comment by John1231983** [comment]
@Banyueqin : Thanks. But this is result after fix it
```

2627.982177734375
749.9542846679688
538.127685546875
401.61016845703125
308.06640625
241.04234313964844
191.45272827148438
153.8364715576172
124.77854919433594
102.02399444580078
84.0322265625
69.68601989746094
58.185302734375
48.94459533691406
41.494354248046875
35.47389221191406
30.60963249206543
26.680410385131836
23.508365631103516
20.950298309326172
tensor([8.7028]) tensor([5.3088]) 18.911481857299805

```

**Comment by Banyueqin** [comment]
@John1231983  
![image](https://user-images.githubusercontent.com/32731721/60241661-6a9bb480-98e6-11e9-8f6b-58534eb60a30.png)

result:
![image](https://user-images.githubusercontent.com/32731721/60241698-7c7d5780-98e6-11e9-9c3b-151bec3e3bbd.png)

**Comment by John1231983** [comment]
No. We expect result is 10 and 0.

**Comment by lilaczhou** [comment]
@hardianlawi  My reuslt goes to negative number after some steps,and I think it results from log_var(sigma),would you mind give me some suggestions?
![image](https://user-images.githubusercontent.com/45725450/63644626-c5536000-c71f-11e9-8390-24b6e187c688.png)
